tests/lib/external/snapd-testing-tools/utils/spread-filter

```diff
@@ -0,0 +1,223 @@
+
+    def _get_parent_with_task(self, change_path: str) -> str:
+        if self.verbose:
+            print("calculating parent with task {}".format(change_path))
+        parent_path = os.path.dirname(change_path)
+        while parent_path != "":
+            if os.path.isfile(os.path.join(parent_path, self.TASK)):
+                return parent_path
+            parent_path = os.path.dirname(parent_path)
+        return ""
+
+    def _clean_executions(self, executions: list[str]) -> list[str]:
+        if self.verbose:
+            print("cleaning executions for {}".format(executions))
+
+        final_executions = []
+        # Get the execution paths ordered starting from the longest one
+        sorted_executions = sorted(executions, key=len, reverse=True)
+        for i in range(len(sorted_executions)):
+            discard_execution = False
+            long_execution = sorted_executions[i]
+            if self.verbose:
+                print("  checking repeated executions for {}".format(long_execution))
+
+            for j in range(i + 1, len(sorted_executions)):
+                short_execution = sorted_executions[j]
+                if self.verbose:
+                    print(
+                        "    comparing repeated executions for {}".format(
+                            short_execution
+                        )
+                    )
+
+                # if the shorter executions is a test, we continue
+                if not short_execution.endswith("/"):
+                    if long_execution == short_execution:
+                        discard_execution = True
+                        break
+                # if the shorter execution is a either suite or group of suites
+                else:
+                    # if the long executions starts with short execution means
+                    # the long execution is included in the short one and it can
+                    # be discarded
+                    if long_execution.startswith(short_execution):
+                        discard_execution = True
+                        break
+
+            # take decision after all shorter executions have been compared
+            if not discard_execution:
+                final_executions.append(long_execution)
+
+        return final_executions
+
+    # Retrieves the list of tests to execute
+    def get_executions(self, changes: list[str]) -> list[str]:
+        all_executions = []
+        for change in changes:
+            executions = self.rules.calc_executions(change)
+            # When the tests to run as tagged as SELF means the current test
+            # has to be executed
+            for execution in executions:
+                if execution == self.SELF:
+                    self_execution = self._calc_self(change)
+                    if self_execution != "":
+                        all_executions.append(self_execution)
+                # When the tests to run as tagged as NONE means that no tests
```

Comment:
When the tests to run `are` tagged as NONE`,` it means that no tests
`should` be executed.